cli/internal/cmd/upgradeapply.go

```diff
@@ -149,7 +149,7 @@ func (u *upgradeApplyCmd) upgradeApply(cmd *cobra.Command, upgradeDir string, fl
 			}
 		}
 	}
-	validK8sVersion, err := validK8sVersion(cmd, conf.KubernetesVersion, flags.yes)
+	conf.KubernetesVersion, err = validK8sVersion(cmd, string(conf.KubernetesVersion), flags.yes)
```

Comment:
This should be persisted, shouldn't it?

Reply:
why should it be persisted? If a user specifies 'v1.27' I'm not sure if we should add the patch version. He might want to always have the latest patch version. Otherwise I would expect that he sets the complete version (e.g. through config generate)